src/blueapi/log.py

```diff
@@ -0,0 +1,85 @@
+import logging
+import os
+
+from graypy import GELFTCPHandler
+
+from blueapi.config import LoggingConfig
+
+
+class InstrumentTagFilter(logging.Filter):
+    """Filter to attach instrument and beamline attributes to LogRecords.
+
+    Attaches attributes `instrument` and `beamline` to all LogRecords that are passed
+    through. Value is taken from `BEAMLINE` env var, defaulting to `dev`.
+    """
+
+    instrument: str | None = os.environ.get("BEAMLINE", "dev")
```

Comment:
```suggestion
    instrument: str = os.environ.get("BEAMLINE", os.environ.get("INSTRUMENT", "dev"))
```

Gets us:
1. `$BEAMLINE` if it exists
2. `$INSTRUMENT` if if exists
3. "dev"

Which therefore is never None and we can tighten the bounds on this.

Reply:
Implemented